### PR TITLE
Tweak OutdatedWarning.generate docs

### PR DESCRIPTION
**Pull request by pfitzseb**

### CODE REVIEW

File: src/OutdatedWarning.jl [hunk 1]
```diff
@@ -135,12 +135,14 @@ end
...
 
-This function adds a warning (and `noindex` meta tag) to all versions of
-the documentation in `root`.
+This function adds a (nonconditional) warning (and `noindex` meta tag) to all
+versions of the documentation in `root`.
 
 `force` overwrites a previous injected warning message created by this function.
 
-A typical use case is to run this on the `gh-pages` branch of a packge.
+A typical use case is to run this on the `gh-pages` branch of a packge. Make sure
```
Comment (fredrikekre) [inline src/OutdatedWarning.jl:143]:
```suggestion
A typical use case is to run this on the `gh-pages` branch of a package. Make sure
```

File: src/OutdatedWarning.jl [hunk 2]
```diff
@@ -135,12 +135,14 @@ end
...
+versions of the documentation in `root`.
 
 `force` overwrites a previous injected warning message created by this function.
 
-A typical use case is to run this on the `gh-pages` branch of a packge.
+A typical use case is to run this on the `gh-pages` branch of a package. Make sure
+you review which changes you check in if you are *not* tagging a new release
+of your package's documentation at the same time.
 """
 generate(root::String; kwargs...) = generate(stdout, root; kwargs...)
```
Comment (odow) [inline src/OutdatedWarning.jl:147]:
```suggestion
function generate(root::String; kwargs...)
    @warn("""
Adding an `OutdatedWarning` to all versions of the documentation in $(root).

A typical use case is to run this on the `gh-pages` branch of a package. Make sure
you review which changes you check in if you are *not* tagging a new release
of your package's documentation at the same time. For example you should 
_not_ check in the folder `vX.Y.Z` corresponding to your latest release. 
""")
    return generate(stdout, root; kwargs...)
end
```

Any objection to making this an explicit warning? If it's a helper script, a note in the docstring may not be enough for people.

**Review by mortenpi — APPROVED** [review]
LGTM, thanks everyone! I took the liberty of making a couple of tiny fixes to the warning message.

**Comment by mortenpi** [comment]
This should have been merged a long time ago. Sorry about that.

**Pull request merged**